### DISCUSSION TITLE
passing sqlConnections to custom class (.Net Core)

### DISCUSSION THREAD

**Discussion by TheWebbe** [serenity-is/Serenity#6284] [Help] [answered]
I followed the [guides ](https://github.com/serenity-is/Serenity/wiki/Background-Tasks-Using-Hangfire-Background-Jobs) to add Hangfire for background jobs and that appears to be working correctly, though I am running trouble trying to access my database. 

Ideally I would like a background job to run a query and perform some actions.

from the guide (SimpleJob.cs): 
```
using Serenity;
using System;

namespace YourAwesomeProject.Common.Jobs
{
    public class SimpleJob
    {
        public void Run()
        {
            new Exception("Hello Serenity from Hangfire!").Log();
            // ***I would like to query against the database here***

        }
    }
}
```

startup.cs:
```
...
BackgroundJob.Enqueue<Common.Jobs.SimpleJob>(job => job.Run());
RecurringJob.AddOrUpdate<Common.Jobs.SimpleJob>(job => job.Run(), Cron.Hourly);
RecurringJob.AddOrUpdate<Common.Jobs.SimpleJob>(job => job.Run(), "0 * * * *");
...
```

it would appear as though I need to pass sqlConnections into the Run function, but I can't seem to find a way to do that. Any help would be greatly appreciated.

**Comment by edwardch** [comment, marked as answer]
Try something like this

```cs
public class SimpleJob
{
    private readonly ISqlConnections Connections;

    public SimpleJob(ISqlConnections connections) 
    {
        this.Connections = connections ?? throw new ArgumentNullException(nameof(connections));
    }

    public void Run()
    {
        using (var connection = Connections.NewFor<MyRow>())
        {
            // do stuff
        }
    }
//...
```

**Reply by edwardch** [reply]
@TheWebbe I have updated the [wiki section](https://github.com/serenity-is/Serenity/wiki/Background-Tasks-Using-Hangfire-Background-Jobs#5-create-a-simple-job) to include the example.

HTH
